feat: Add printer name to browser tab while printing or complete

mikebcbc opened the pull request:
Adds the printer name to the browser tab during printing and complete statuses.

<!--
  For Work In Progress Pull Requests, please use the Draft PR feature,
  see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.

  For a timely review/response, please avoid force-pushing additional
  commits if your PR already received reviews or comments.

  We use squash-merge to merge PRs, so the commit history is clean, but we use the PR title as the commit message.
  So we recommend you to use the PR title with conventional commits type prefixes. We use the following prefixes for the PR title:
    - feat: A new feature
    - fix: A bug fix
    - docs: Documentation only changes
    - style: Changes that do not affect the meaning of the code (white-space, formatting, etc)
    - refactor: A code change that neither fixes a bug nor adds a feature
    - perf: A code change that improves performance
    - test: Adding missing tests or correcting existing tests
    - chore: Changes to the build process or auxiliary tools and libraries such as documentation generation
    - locale: Changes to the translations

  You can found more information about Conventional Commits here: https://www.conventionalcommits.org/en/v1.0.0/.

  Before submitting a Pull Request, please ensure you've done the following:
  - 📖 Read the Mainsail Contributing Guidelines: https://github.com/mainsail-crew/mainsail/blob/HEAD/CONTRIBUTING.md#-submitting-a-pull-request-pr.
  - 📖 Read the Mainsail Code of Conduct: https://github.com/mainsail-crew/mainsail/blob/HEAD/.github/CODE_OF_CONDUCT.md.
  - 👷‍♀️ Create small PRs. In most cases, this will be possible.
  - ✅ Provide tests for your changes.
  - 📝 Use descriptive commit messages.
  - 📗 Update any related documentation and include any relevant screenshots.
-->

## Description

This PR addresses #1322 and adds the printer name to the browser tab while the printer is printing, or when the print is complete.

<!--
Please do not leave this blank
This PR [adds/removes/fixes/replaces] the [feature/bug/etc].
-->

## Related Tickets & Documents

<!--
Please use this format link issue numbers: Fixes #123
https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword
-->

#1322 

## Mobile & Desktop Screenshots/Recordings

<img width="261" alt="image" src="https://user-images.githubusercontent.com/15934114/235999802-950298b4-6cdb-460d-b12a-ac7bb984c685.png">

## [optional] Are there any post-deployment tasks we need to perform?

<!-- note: PRs with deleted sections will be marked invalid -->

No post-deployment tasks.

meteyou commented on src/components/settings/SettingsUiSettingsTab.vue:
```diff
@@ -341,7 +341,7 @@ export default class SettingsUiSettingsTab extends Mixins(BaseMixin) {
 
     get disableFanAnimation() {
-        return this.$store.state.gui.uiSettings.hideSaveConfigForBedMash ?? false
+        return this.$store.state.gui.uiSettings.disableFanAnimation ?? false
```
please create a separate PR for this bugfix

mikebcbc replied:
done: https://github.com/mainsail-crew/mainsail/pull/1381

meteyou commented on src/locales/en.json:
```diff
@@ -49,12 +49,12 @@
         },
         "Titles": {
-            "Complete": "Complete - {filename}",
+            "Complete": "Complete - {filename} - {name}",
```
i would recommend to change the name of these translations, that all translators see a new key to translate

mikebcbc replied:
Can you give an example of what you mean? Do you want me to go through every translation and update? Or change the key name? Won't that break the other translations if I do the second one?

meteyou replied:
That's what changing the key is precisely about. Just as an example:

You only change the English translation here. The key remains the same, so no translator is made aware that there are new translations that are not translated. If we take DE as an example, without this change, the old text is still displayed without the name.

So it is better to change the key, which does not exist as a translation, but it has a correct fallback in English with all variables or texts it should have.

An working "new" key would be "CompleteWithPrintername"...

mikebcbc replied:
Ahh, understood! That makes sense - I went ahead and updated this! Thanks for clarifying!

meteyou commented:
Thx for your PR. I have only one question about it. What happens when the user don't set a printername in the settings?

mikebcbc commented:
> Thx for your PR. I have only one question about it. What happens when the user don't set a printername in the settings?

@meteyou - good question, I updated this PR to handle that. Before it would just show the hyphen, but now I am conditionally showing the hyphen. Take a look at the PR. Here's two examples:

printername defined:
<img width="254" alt="image" src="https://user-images.githubusercontent.com/15934114/236699798-98cbb7c7-7af3-4a7e-a722-0e8f5e72c96c.png">

printername not defined:

<img width="253" alt="image" src="https://user-images.githubusercontent.com/15934114/236699817-05fa1fab-5189-49f0-b47b-4356e768a198.png">

meteyou commented:
hmm... maybe my "idea" with adding the `{name}` in the translation itself was wrong for this use case. maybe you should just use getter to add " - <name>", when it exists and we also don't have an issue with the "new" translation keys.

mikebcbc commented:
> hmm... maybe my "idea" with adding the `{name}` in the translation itself was wrong for this use case. maybe you should just use getter to add " - ", when it exists and we also don't have an issue with the "new" translation keys.

@meteyou Are you saying to force it in the getter? Like this:

```vue
return i18n.t('App.Titles.PrintingETA', {
                    percent: percent,
                    filename: state.printer?.print_stats?.filename,
                    eta
                }) + state.gui?.general.printername ? `- ${state.gui?.general.printername}` : ""
``` 

If so, I feel like that's not very clean... I feel like it would be cleaner to just supply the `- {name}` and update the translation keys. If we don't want to worry about translation, since it's all the same no matter the language I can just update every translation and append this where it fits... what are you thoughts on that?

meteyou commented:
i would use write it a little bit different (just for readability):
```ts
let output = i18n.t('App.Titles.PrintingETA', {
	                    percent: percent,
	                    filename: state.printer?.print_stats?.filename,
	                    eta
	                })

// add printer name if it exists
if (state.gui?.general.printername) output += ` - ${state.gui.general.printername}`

return output
```

I just don't like to add "decorators" in a translation variable.

just my 2 cents about this topic...
(my solution have not to be the best way... its just my style to do it...)

mikebcbc commented:
> 

@

> i would use write it a little bit different (just for readability):
> 
> ```ts
> let output = i18n.t('App.Titles.PrintingETA', {
> 	                    percent: percent,
> 	                    filename: state.printer?.print_stats?.filename,
> 	                    eta
> 	                })
> 
> // add printer name if it exists
> if (state.gui?.general.printername) output += ` - ${state.gui.general.printername}`
> 
> return output
> ```
> 
> I just don't like to add "decorators" in a translation variable.
> 
> just my 2 cents about this topic... (my solution have not to be the best way... its just my style to do it...)

@meteyou - I think that makes sense. I think it does all come down to style - looks clean to me. I updated this PR and removed it from the translation keys. Let me know what you think! Thanks for the clarification!

mikebcbc commented:
@meteyou - anything else you need from me on this PR, or is it good to go?

meteyou commented:
LGTM! Thanks!

wikolii commented:
Hi, is there any way one could choose to have the machine name in front of the tab description?
In my case I normally just print the same part in many printers and having to wait for the popup with the full tab description to know the machine name is time consuming.

I would even choose not to have the filename on the tab description, maybe there could be some setting for that.
Thanks!
